pkg/playback/http_recorder.go

```diff
@@ -0,0 +1,143 @@
+		return
+	}
+	wrappedResp, err := newHTTPResponse(resp)
+	if err != nil {
+		writeErrorToHTTPResponse(w, fmt.Errorf("Unexpected error forwarding webhook to client: %w", err), 500)
+		return
+	}
+
+	// --- Write response back to client
+
+	// We defer writing anything to the response until their final values are known, since certain fields can
+	// only be written once. (golang's implementation streams the response, and immediately writes data as it is set)
+	if err != nil {
+		writeErrorToHTTPResponse(w, fmt.Errorf("Unexpected error forwarding webhook to client: %w", err), 500)
+		return
+	}
+	err = httpRecorder.recorder.write(incomingInteraction, wrappedReq, wrappedResp)
+	if err != nil {
+		writeErrorToHTTPResponse(w, fmt.Errorf("Unexpected error writing webhook interaction to cassette: %w", err), 500)
+		return
+	}
+
+	// Now we can write to the response:
+	// The header *must* be written first, since writing the body with implicitly and irreversibly set
+	// the status code to 200 if not already set.
+	copyHTTPHeader(w.Header(), wrappedResp.Headers) // header map must be written before calling w.WriteHeader
+	w.WriteHeader(wrappedResp.StatusCode)
+	io.Copy(w, bytes.NewBuffer(wrappedResp.Body))
+}
+
+func (httpRecorder *recordServer) handler(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("\n--> %v to %v", r.Method, r.RequestURI)
+
+	wrappedReq, err := newHTTPRequest(r)
+	if err != nil {
+		writeErrorToHTTPResponse(w, fmt.Errorf("Unexpected error processing incoming webhook request: %w", err), 500)
+		return
+	}
+
+	// --- Pass request to remote
+	var resp *http.Response
+
+	resp, err = forwardRequest(&wrappedReq, httpRecorder.remoteURL+r.RequestURI)
+	if err != nil {
+		writeErrorToHTTPResponse(w, err, 500)
+		return
+	}
+
+	wrappedResp, err := newHTTPResponse(resp)
+	if err != nil {
+		writeErrorToHTTPResponse(w, err, 500)
+		return
+	}
+
+	fmt.Printf("\n<-- %v from %v\n", resp.Status, strings.ToUpper(httpRecorder.remoteURL))
+
+	// --- Write response back to client
+
+	// We defer writing anything to the response until their final values are known, since certain fields can
+	// only be written once. (golang's implementation streams the response, and immediately writes data as it is set)
+	if err != nil {
+		writeErrorToHTTPResponse(w, err, 500)
+		return
+	}
+	err = httpRecorder.recorder.write(outgoingInteraction, wrappedReq, wrappedResp)
+	if err != nil {
+		writeErrorToHTTPResponse(w, fmt.Errorf("Error when recording HTTP response to cassette: %w", err), 500)
+		return
+	}
+	// Now we can write to the response:
+	// The header *must* be written first, since writing the body with implicitly and irreversibly set
+	// the status code to 200 if not already set.
+	copyHTTPHeader(w.Header(), wrappedResp.Headers) // header map must be written before calling w.WriteHeader
+	w.WriteHeader(wrappedResp.StatusCode)
+	io.Copy(w, bytes.NewBuffer(wrappedResp.Body))
```

Comment:
are there any opportunities for the cli to panic if any of these lines fail here?

Reply:
Oh good catch - I added logging and now have the code check the error returned by `io.Copy` and log.Fatal if non-`nil` . feacb33396bd5fb09a3feef1749ea5d61eba03c3